### PR TITLE
Tests: grpconv wants to take all memory if /etc/group contains duplic…

### PR DESCRIPTION
**Pull request by aborah-sudo**
…ate entries

https://bugzilla.redhat.com/show_bug.cgi?id=507706
https://bugzilla.redhat.com/show_bug.cgi?id=494575

### CODE REVIEW

**Review by ikerexxe — CHANGES_REQUESTED** [review]
The bugzilla ticket that you mention in the code and the PR comment doesn't match with what you are testing. Can you review it and provide the correct link?

**Comment by aborah-sudo** [comment]
> The bugzilla ticket that you mention in the code and the PR comment doesn't match with what you are testing. Can you review it and provide the correct link?

Fixed now

File: multihost_test/Bugzillas/test_Regression.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          4. The groups with the name pattern group_ from /tmp/anuj are appended back into the /etc/group file.
+            This effectively doubles the number of such groups.
+          5. Checks again and asserts that the number of groups with the name pattern group_ in
+            the /etc/group file should now be 20.
+          6. The grpconv command is executed in the background (indicated by &).
+            This command initializes the shadow file.
+          7. Checks if the grpconv command is still running. If it is not running,
+            a subprocess.CalledProcessError exception will be raised.
+          8. Kill the grpconv process, if it's still running. Again, if the process doesn't exist,
+            a subprocess.CalledProcessError exception will be raised.
```
Comment (ikerexxe) [inline multihost_test/Bugzillas/test_Regression.py:41]: Do you know why we need to do this?

**Review by ikerexxe — COMMENTED** [review]
Just a minor comment

File: multihost_test/Bugzillas/test_Regression.py (hunk 1 above)
Comment (aborah-sudo) [reply multihost_test/Bugzillas/test_Regression.py:41]: Just a double check , so that the process is not running

Comment (ikerexxe) [reply multihost_test/Bugzillas/test_Regression.py:41]: And why we need to check? Wouldn't it be better to call `grpconv` synchronously and wait until it finishes?

Comment (aborah-sudo) [reply multihost_test/Bugzillas/test_Regression.py]: ok, i have removed this part .

Comment (ikerexxe) [reply multihost_test/Bugzillas/test_Regression.py]: With this comment I meant that step 6 should run `grpconv` in the foreground, then steps 7 and 8 wouldn't be necessary

Comment (aborah-sudo) [reply multihost_test/Bugzillas/test_Regression.py]: changes are done

File: multihost_test/Bugzillas/test_Regression.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          8. Should succeed
+        """
+        for i in range(10):
+            execute_cmd(multihost, f"groupadd group_{i}")
+        assert execute_cmd(multihost, "grep group_ /etc/group | wc -l").stdout_text.split('\n')[0] == '10'
+        execute_cmd(multihost, "cat /etc/group > /tmp/anuj")
+        execute_cmd(multihost, "grep group_ /tmp/anuj >> /etc/group")
+        assert execute_cmd(multihost, "grep group_ /etc/group | wc -l").stdout_text.split('\n')[0] == '20'
+        execute_cmd(multihost, "grpconv &")
+        execute_cmd(multihost, "sleep 20")
```
Comment (ikerexxe) [inline multihost_test/Bugzillas/test_Regression.py:59]: In order to call `grpconv` in the foreground you need to remove the `&`. Once that is done the call should be done synchronously, which means that it won't return until it has finished. If that's the case, then you can also remove the `sleep`.

File: multihost_test/Bugzillas/test_Regression.py (hunk 1 above)
Comment (ikerexxe) [reply multihost_test/Bugzillas/test_Regression.py]: I don't think we are understanding each other, so I added a new comment inline.

File: multihost_test/Bugzillas/test_Regression.py (hunk 2 above)
Comment (aborah-sudo) [reply multihost_test/Bugzillas/test_Regression.py:59]: Fixed now

**Review by ikerexxe — APPROVED** [review]
LGTM! Thanks for taking your work in porting these test cases

**Pull request merged**